lib/app/cli.ts

```diff
@@ -0,0 +1,249 @@
+    static?: string;
+    local: boolean;
+    version: boolean;
+}
+
+/**
+ * Parse command-line arguments and return parsed options
+ * @param argv The command-line arguments to parse
+ */
+export function parseCommandLine(argv: string[]): CompilerExplorerOptions {
+    const program = new Command();
+    program
+        .name('compiler-explorer')
+        .description('Interactively investigate compiler output')
+        .option('--env <environments...>', 'Environment(s) to use', ['dev'])
+        .option('--root-dir <dir>', 'Root directory for config files', './etc')
+        .option('--host <hostname>', 'Hostname to listen on')
+        .option('--port <port>', 'Port to listen on', parseNumberForOptions, 10240)
+        .option('--prop-debug', 'Debug properties')
+        .option('--debug', 'Enable debug output')
+        .option('--dist', 'Running in dist mode')
+        .option('--no-remote-fetch', 'Ignore fetch marks and assume every compiler is found locally')
+        .option('--tmpDir, --tmp-dir <dir>', 'Directory to use for temporary files')
+        .option('--wsl', 'Running under Windows Subsystem for Linux')
+        .option('--language <languages...>', 'Only load specified languages for faster startup')
+        .option('--no-cache', 'Do not use caching for compilation results')
+        .option('--ensure-no-id-clash', "Don't run if compilers have clashing ids")
+        .option('--logHost, --log-host <hostname>', 'Hostname for remote logging')
+        .option('--logPort, --log-port <port>', 'Port for remote logging', parseNumberForOptions)
+        .option('--hostnameForLogging, --hostname-for-logging <hostname>', 'Hostname to use in logs')
+        .option('--suppressConsoleLog, --suppress-console-log', 'Disable console logging')
+        .option('--metricsPort, --metrics-port <port>', 'Port to serve metrics on', parseNumberForOptions)
+        .option('--loki <url>', 'URL for Loki logging')
+        .option('--discoveryonly, --discovery-only <file>', 'Output discovery info to file and exit')
+        .option('--prediscovered <file>', 'Input discovery info from file')
+        .option('--static <dir>', 'Path to static content')
+        .option('--no-local', 'Disable local config')
+        .option('--version', 'Show version information');
+
+    program.parse(argv);
+    return program.opts() as CompilerExplorerOptions;
+}
+
+/**
+ * Extract git release information from repository or file
+ */
+export function getGitReleaseName(distPath: string, isDist: boolean): string {
+    // Use the canned git_hash if provided
+    const gitHashFilePath = path.join(distPath, 'git_hash');
+    if (isDist && fs.existsSync(gitHashFilePath)) {
+        return fs.readFileSync(gitHashFilePath).toString().trim();
+    }
+
+    // Check if we have been cloned and not downloaded
+    if (fs.existsSync('.git')) {
+        return child_process.execSync('git rev-parse HEAD').toString().trim();
+    }
+
+    // unknown case
+    return '';
```

Comment:
Maybe we could add a unique name like "<no git hash found>" instead of having it be completely invisible?

Reply:
Done